commons/zenoh-shm/src/api/provider/types.rs

```diff
@@ -84,11 +86,13 @@ impl AllocAlignment {
     /// assert_eq!(aligned_size, 8);
     /// ```
     #[zenoh_macros::unstable_doc]
-    pub fn align_size(&self, size: usize) -> usize {
+    pub fn align_size(&self, size: NonZeroUsize) -> NonZeroUsize {
         let alignment = self.get_alignment_value();
-        match size % alignment {
+        match size.get() % alignment {
             0 => size,
-            remainder => size + (alignment - remainder),
+            remainder => unsafe {
+                NonZeroUsize::new_unchecked(size.get() + (alignment.get() - remainder))
```

Comment:
What's the guarantee of this `unsafe` block?

Reply:
@Mallets this unsafe block is always safe:
1. remainder is always less than alignment
2. because of 1, the value of (alignment.get() - remainder) is always > 0
3. because of 2, we add nonzero size to nonzero (alignment.get() - remainder)
4. we make sure that there is no overflow condition in 3 by means of alignment limitation in `new` by limiting pow value

I will add a comment to the code

@wyfo unfortunately, the compiler is not able to infer this!

Reply:
Indeed, it's unfortunate, and I'm quite surprise that it's not able to even infer that `alignment.get() - remainder` is non-zero 🤔